src/number.cr

```diff
@@ -389,19 +389,76 @@ struct Number
     self.class.new((x / y).round * y)
   end
 
-  # Rounds this number to a given precision in decimal *digits*.
+  # Rounds this number to a given precision.
+  #
+  # Rounds to the specified number of *digits* after the decimal place,
+  # (or before if negative), in base *base*.
+  #
+  # The rounding *mode* controls the direction of the rounding. The default is
+  # `RoundingMode::TIES_AWAY` which rounds to the nearest integer, with ties
+  # (fractional value of `0.5`) being rounded away from zero.
   #
   # ```
   # -1763.116.round(2) # => -1763.12
   # ```
-  def round(digits = 0, base = 10)
-    x = self.to_f
+  def round(digits : Number, base = 10, *, mode : RoundingMode = :ties_away)
+    if digits < 0
+      multiplier = base.to_f ** digits.abs
+      shifted = self / multiplier
+    else
+      multiplier = base.to_f ** digits
+      shifted = self * multiplier
+    end
+
+    rounded = shifted.round(mode)
+
     if digits < 0
-      y = base.to_f ** digits.abs
-      self.class.new((x / y).round * y)
+      result = rounded * multiplier
     else
-      y = base.to_f ** digits
-      self.class.new((x * y).round / y)
+      result = rounded / multiplier
+    end
+
+    self.class.new result
+  end
+
+  # Specifies rounding behaviour for numerical operations capable of discarding
+  # precision.
+  enum RoundingMode
```

Comment:
I'd move it to the beginning of the file for a better discover-ability.

Reply:
What difference does it make? I think it should stay in proximity to the methods that use it.

Reply:
We could even consider extracting all features related to rounding into a separate file.

Reply:
> We could even consider extracting all features related to rounding into a separate file. 

Sounds even better.

Reply:
Let's leave that for a follow-up.